docs/developer-docs/updates/release-notes/release-notes.md

```diff
@@ -1,5 +1,83 @@
 # Release Notes
 
+# What’s new in 0.10.1
+
+## Changes to DFX
+
+### Fixed: Webpack config no longer uses CopyPlugin
+
+Dfx already points to the asset canister's assets directory, and copying to disk could sometimes
+lead to an annoying "too many open files" error.
+
+### Fixed: HSMs are once again supported on Linux
+
+On Linux, dfx 0.10.0 failed any operation with an HSM with the following error:
+Error: IO: Dynamic loading not supported
+
+The fix was to once again dynamically-link the Linux build.
+
+### New feature: error explanation and fixing instructions engine
+
+Dfx is now capable of providing explanations and remediation suggestions for entire categories of errors at a time.
+Explanations and suggestions will slowly be added over time.
+To see an example of an already existing suggestion, run `dfx deploy --network ic` while using an identity that has no wallet configured.
+
+### Improvement: add context to errors
+
+Most errors that happen within dfx are now reported in much more detail. No more plain `File not found` without explanation what even was attempted.
+
+### Fixed: identities with configured wallets are not broken anymore and removed only when using the --drop-wallets flag
+
+When an identity has a configured wallet, dfx no longer breaks the identity without actually removing it.
+Instead, if the --drop-wallets flag is specified, it properly removes everything and logs what wallets were linked,
+and when the flag is not specified, it does not remove anything.
+
+The behavior for identities without any configured wallets is unchanged.
+
+### New Feature: print wallet balance in a human readable form
+
+Default behaviour changed for `+dfx wallet balance+`, it will now print cycles amount upscaled to trillions.
+
+New flag `+--precise+` added to `+dfx wallet balance+`. Allows to get exact amount of cycles in wallet (without upscaling).
```

Comment:
```suggestion
Default behaviour changed for `dfx wallet balance`, it will now print cycles amount upscaled to trillions.

New flag `--precise` added to `dfx wallet balance`. Allows to get exact amount of cycles in wallet (without upscaling).
```
markdown syntax